src/pages/OfficeView.tsx

```diff
@@ -1,46 +1,37 @@
-import { useCallback, useEffect, useState } from 'react'
+import { useState } from 'react'
 import { Card, Col, FormControl, InputGroup, Row } from 'react-bootstrap'
 import { Link, useLoaderData, useNavigate } from 'react-router-dom'
 import { useTranslation } from 'react-i18next'
 import OfficeInfo from 'src/components/OfficeInfo'
 import RatingsBadge from 'src/components/RatingsBadge'
 import { OfficeLoaderReturn } from 'src/loaders/officeLoader'
-import Agent from 'src/types/Agent'
 import businessIcon from 'src/images/business.svg'
 import { ApiAgent } from 'src/api'
-import Paginator from 'src/components/Paginator'
-import { debounce } from 'lodash-es'
-import SearchMeta from 'src/types/SearchMeta'
+import { useSuspenseInfiniteQuery } from '@tanstack/react-query'
+import AnimatedList from 'src/components/AnimatedList'
+import { InView } from 'react-intersection-observer'
 
 export default function OfficeView() {
   const { office } = useLoaderData() as OfficeLoaderReturn
   const [searchValue, setSearchValue] = useState('')
   const navigate = useNavigate()
   const { t } = useTranslation()
-  const [agents, setAgents] = useState<Agent[]>([])
-  const [meta, setMeta] = useState<SearchMeta>()
 
-  const getAgents = useCallback(
-    async (page: number) => {
-      return await ApiAgent.list({
-        officeId: office.id,
-        search: searchValue,
-        page,
-      })
-    },
-    [office.id, searchValue],
-  )
-
-  useEffect(() => {
-    const initializeAgents = async (page: number) => {
-      const agentData = await getAgents(page)
-      setAgents(agentData.data)
-      setMeta(agentData.meta)
-    }
-    const handleSearchInput = debounce(initializeAgents, 500)
+  const { data, fetchNextPage, hasNextPage, isFetching } =
+    useSuspenseInfiniteQuery({
+      queryKey: ['searchVal', searchValue],
```

Comment:
for the first part of the key, use a descriptive name that describes the dataset. I know "search" seemed generic, but it was actually named that due to being on the "Search" page. Something like `officeAgents` would be good here. The second part, `searchValue` is perfect.

Reply:
Started with 'agents' as key but thought it was supposed to be similar to the variable name. Great tip! Change incorporated.

Reply:
@eddie-m-m I just noticed a big here where I'm seeing the same list of agents for each office. We need to pass in the `office.id` as part of the queryKey as well.